include/mbedtls/cipher.h

```diff
@@ -34,7 +34,7 @@
 #include "mbedtls/platform_util.h"
 
 #if defined(MBEDTLS_GCM_C) || defined(MBEDTLS_CCM_C) || defined(MBEDTLS_CHACHAPOLY_C)
-#define MBEDTLS_CIPHER_MODE_AEAD
+#define MBEDTLS_CIPHER_HAVE_AEAD
```

Comment:
[This should have been part of my review but for some reason github didn't include it.]
The previous name was `MODE_AEAD`, not `HAVE`...

Reply:
Should I also replace the new `MBEDTLS_CIPHER_HAVE_SOME_AEAD_VIA_LEGACY_OR_USE_PSA` with `MBEDTLS_CIPHER_MODE_AEAD_VIA_LEGACY_OR_USE_PSA`? They are supposed to be similar with only the `USE_PSA` option to differentiate them